csi/controller_server.go

```diff
@@ -442,21 +442,17 @@ func (cs *ControllerServer) ControllerPublishVolume(ctx context.Context, req *cs
 
 	return cs.publishVolume(volume, nodeID, attachmentID, func() error {
 		checkVolumePublished := func(vol *longhornclient.Volume) bool {
-			if isVolumeShareAvailable(vol) {
-				return true
-			}
-			attachment, ok := vol.VolumeAttachment.VolumeAttachmentStatus[attachmentID]
-			if !ok {
-				return false
-			}
-			return attachment.AttachError == "" && attachment.Attached && isEngineOnNodeAvailable(vol, nodeID)
+			attachment, ok := vol.VolumeAttachment.Attachments[attachmentID]
+			return ok && attachment.Satisfied
```

Comment:
Check `Generation` here

Reply:
We already check it at [here](https://github.com/longhorn/longhorn-manager/pull/1541/files#diff-22d9efdab1c8ceb9165fd0f1b5777cec7d012c74ec1d1f1d84c9e96d918b6fc0R1303) before returning data back to the client